### PR TITLE
feat: adding playlist page and its functionality

### PR DESCRIPTION
**Pull request by mariyasada**
-user can create a playlist and add the video to a particular playlist
-delete a video from a particular playlist
-delete a particular playlist

https://user-images.githubusercontent.com/77275100/164885837-5250ae46-5f70-4261-b559-c7be04414b09.mp4

### CODE REVIEW

File: src/Components/playListModal/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                  onChange={(e) => {
+                    if (e.target.checked) {
+                      addVideoToPlayList(video, playlist._id, playlist.title);
+                      setShowModal(false);
+                      setIsOpen(false);
+                    } else {
+                      deleteVideoFromPlayList(video, playlist._id);
+                      setShowModal(false);
+                      setIsOpen(false);
+                    }
```
Comment (JantuDeb) [inline src/Components/playListModal/Modal.jsx:41]: define separate function for this 
```
if(){}
else{}
setShowModal(false);
setIsOpen(false);
```

File: src/Components/popupbox/WatchLater.jsx [hunk 2]
```diff
@@ -40,6 +53,7 @@ export const WatchLaterBox = ({ video }) => {
             onClick={() => {
               if (loginStatus) {
                 addToWatchLaterVideo(video);
+                setIsOpen(false);
               } else {
                 toast("please login to continue", { icon: "✔️" });
                 navigate("/loginpage");
```
Comment (JantuDeb) [inline src/Components/popupbox/WatchLater.jsx:59]: same here

**Review by JantuDeb — APPROVED** [review]
Overall this looks fine, `remove console.log`

File: src/Components/popupbox/WatchLater.jsx [hunk 3]
```diff
@@ -21,15 +28,21 @@ export const WatchLaterBox = ({ video }) => {
   const {
     user: { loginStatus },
   } = useAuth();
+  const { playlistState } = usePlayList();
   const { likedList, watchLaterList } = videoState;
+  const { playList } = playlistState;
+  console.log(playlistState, "watch");
```
Comment (tusharkandpal) [inline src/Components/popupbox/WatchLater.jsx:34]: Remove this log please

File: src/Context/Reducer/likedvideosReducer.js [hunk 4]
```diff
@@ -1,35 +1,33 @@
-export const likedvideosReducer = (state,action) => {
+export const likedvideosReducer = (state,action) => {   
    switch (action.type) {
        case "LOAD_LIKED_VIDEOS":
-           return {...state, likedList:[...action.payload]}
+           return {...state, likedList:action.payload}
```
Comment (tusharkandpal) [inline src/Context/Reducer/likedvideosReducer.js:4]: Your `payload` should be an object.

File: src/Services/playListServices/CreatePlaylistServices.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import axios from "axios";
+
+export const createPlayListService=async(user,title)=>{
+    try {
+        const {data}=await axios.post("/api/user/playlists",{playlist:{title:title,description:"dummy text"}},{
+            headers:{
+             authorization: user.authenticationToken
+            }
+        })    
+        console.log(data);
```
Comment (tusharkandpal) [inline src/Services/playListServices/CreatePlaylistServices.js:10]: Remove this logger too

File: src/Services/playListServices/addVideoToPlayList.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+export const addVideoToPlayListService =async(video,playlistId,user)=>{
+    console.log(playlistId,"from adding service");
+ try { 
+     const {data,status}=await axios.post(`/api/user/playlists/${playlistId}`,{video},{
+         headers:{
+             authorization: user.authenticationToken
+            }
+        })    
+        console.log(data,"from addvideotoplaylist api calls");
```
Comment (tusharkandpal) [inline src/Services/playListServices/addVideoToPlayList.js:11]: Here too!

File: src/index.css [hunk 7]
```diff
@@ -10,6 +10,7 @@ body {
   -webkit-font-smoothing: antialiased;
   -moz-osx-font-smoothing: grayscale;
   font-family: "Roboto Condensed", sans-serif;
+  /* background-color: #0e0d0d; */
```
Comment (tusharkandpal) [inline src/index.css:13]: +1

**Review by tusharkandpal — APPROVED** [review]
Code is well structured. Need a bit of formatting.

**Pull request merged**